libdnf/repo/solvable/Dependency.cpp

```diff
@@ -0,0 +1,165 @@
+#include "Dependency.hpp"
+#include "libdnf/utils/utils.hpp"
+
+#include <vector>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+#   include <solv/pool_parserpmrichdep.h>
+#ifdef __cplusplus
+}
+#endif
+
+
+#define NUMBER_OF_ITEMS_IN_REL_DEPENDENCY 3
+
+class SolvableDependency::Impl {
+public:
+    Impl();
+    Impl(const Impl &source);
+    ~Impl() = default;
+private:
+    friend class SolvableDependency;
+
+    void parseAndCreateDependency(DnfSack *sack, const std::string &dependency);
+    void createRelationalDependency(DnfSack *sack, const std::vector<std::string> &results);
+    bool isRichDependency(const std::string &dependency) const;
+    int determineComparisonType(const std::string &type);
+
+    Id libsolvableId;
+    Pool *pool; /** compatibility */
```

Comment:
Pool is part of DnfSack

Reply:
From `dnf-reldep.h`
```
dnf_reldep_from_pool (Pool *pool, Id r_id);
```
If you tell me how to transform Pool to DnfSack, I am more than happy to remove the attribute.